generator.rb

```diff
@@ -0,0 +1,111 @@
+def is_vowel(letter)
+  case letter
+  when "a", "e", "i", "o", "u"
+    return "true"
+  else
+    return "false"
+  end
+end
+
+def rev_str(strg)
+  aa = strg.split("")
+  rev = Array.new
+  aa.length.times do
+    rev << aa.pop
+  end
+  rev = rev.join
+  return rev
+end
+
+def check_length(word)
+  if word.length < 2
+    until word.length > 1
+      puts "Your word must be 2 or more characters."
+      word = gets.chomp
+    end
+    return word
+  else
+    return word
+  end
+end
+
+def vowel_index(word)
+  count = 0
+  word.each_char { |letter|
+    if is_vowel(letter) == "false"
```

Comment:
In general, I would prefer you use `do..end` over `{..}`. It makes the code a little easier to read.